tech-papers/lbft-verification/lbft-verification.tex

```diff
@@ -0,0 +1,199 @@
+
+The trusted points in history determined by the waypoints can be used in the following situations:
+\begin{itemize}
+\item \textbf{Long range attacks}: trusted recovery relative to a recent point in time for a cold node that does not have an uptodate information locally.
+\item \textbf{Genesis configuration}: a genesis waypoint certifies the genesis \texttt{LedgerInfo} with its initial validator set.
+\item \textbf{Emergency reconfigurations}: consider a disastrous scenario, in which a reconfiguration must be applied without an ability to generate a \texttt{LedgerInfo} with a valid set of signatures (e.g., private keys are lost, or the system needs to be forcefully upgraded to a previous point in time). The \texttt{LedgerInfo} of this reconfiguration cannot be verified via the standard signature checks, and it's going to be verified using waypoints (see more in Section~\ref{waypoint-emergency}).
+\end{itemize}
+
+Obviously, being an off-chain mechanism there is no way to “prove” a waypoint, the nodes choose to trust waypoints for their initialization.
+
+\subsection{Waypoint structure}
+\label{waypoint-structure}
+In order to provide a client with the initial trusted root hash and validators, it is enough for the waypoint to include the information about a LedgerInfo on the epoch boundary (the one that must include the validator set of the next epoch).
+
+The waypoint representation remains a short string that can be easily included in a config or copied from an association website.
+
+\begin{algorithm}[H]
+\myalgorithm
+{
+\Iblock{\rust{Waypoint}}{
+	   \tcp{The version of the reconfiguration transaction that is being approved by this waypoint.}
+        \rust{version: u64}\;
+        \tcp{The hash of the chosen fields of LedgerInfo (including the next validator set).}
+        \rust{value: HashValue}
+}
+}
+\end{algorithm}
+
+
+The fields of a LedgerInfo that are included in the waypoint value are \texttt{epoch}, \texttt{root\_hash}, \texttt{version}, \texttt{timestamp}, and \texttt{next\_validator\_set}. Note that we cannot include all the fields because they might not be deterministic: different clients might observe different values of e.g., \texttt{consensus\_block\_id} of the latest \texttt{LedgerInfo} in an epoch.
+
+The textual representation of a Waypoint is  "version:hex encoding of the root hash". For example, the current testnet genesis waypoint is \newline \texttt{"0:a17035961b97359bcc22b628c1e269f543b265220d22a17e4a5b4e3aed86855b"}.
+
+\subsection{Verifying the latest LedgerInfo with waypoints}
+\label{waypoint-li-verification}
+As described in Section~\ref{verification-epochs}, an \texttt{UpdateToLatestLedgerResponse} message carries a data structure for verifying validator changes, \texttt{ValidatorChangeProof}, which contains a list of all the epoch change \texttt{LedgerInfos}. A client verifies this list in an iterative fashion starting with its initial trusted \texttt{ValidatorSet}.
+
+\begin{figure}[ht]
```

Comment:
it would be helpful if this figure included the information which is used in the text (w_60, w_80, etc.)